tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -484,6 +484,61 @@ public function data_provider_not_supported_webp_images() {
 		yield 'GIFT image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/earth.gif' );
 	}
 
+	/**
+	 * Ensure jpg image is replaced if the request is for the frontend context.
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_not_replace_image_references_when_request_is_for_frontend_context() {
+		$this->mock_action_template_redirect();
+
+		$attachment_id = $this->factory->attachment->create_upload_object(
+			TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg'
+		);
+
+		$tag = wp_get_attachment_image( $attachment_id, 'medium', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+		$this->assertNotSame( $tag, webp_uploads_update_image_references( $tag ) );
+	}
```

Comment:
Let's add one more test that covers the scenario where you're in a frontend context, but before the user facing content (i.e. `template_redirect` has run but `wp_head` hasn't). Maybe that could be called `it_should_not_replace_image_references_when_in_frontend_context_but_before_body`.